CONTRIBUTING.md

```diff
@@ -4,24 +4,52 @@ Thank you for your interest in contributing to our project! 💛
 
 Whether it's a bug report, new feature, correction, or additional documentation, we greatly value feedback and contributions from our community. Please read through these guidelines carefully before submitting a PR or issue and let us know if it's not up-to-date (or even better, submit a PR with your proposed corrections 😉).
 
-# Development Process
+## Getting Started
 
 Our work is done directly on Github and PR's are sent to the GitHub repo by core team members and contributors. Everyone undergoes the same review process to get their changes into the repo.
 
-# Setting up for local development
+This section should get you running with **Amplify Codegen**.
 
-This section should get you running with **Amplify Codegen**. You will need the latest version of [nodejs](https://nodejs.org/en/) on your system and developing locally also requires `yarn` workspaces. You can install it [here](https://classic.yarnpkg.com/en/docs/install#mac-stable).
+### Setting up for local development
 
-Start by, [Forking](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the main branch of [amplify-codegen](https://github.com/aws-amplify/amplify-codegen).
+1. You will need the latest version of [nodejs](https://nodejs.org/en/) on your system and developing locally also requires `yarn` workspaces. You can install it [here](https://classic.yarnpkg.com/en/docs/install#mac-stable).
 
-```
+2. Start by [Forking](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the main branch of [amplify-codegen](https://github.com/aws-amplify/amplify-codegen).
+
+```sh
 $ git clone git@github.com:[username]/amplify-codegen.git
+```
+
+> NOTE: Make sure to always [sync your fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork) with _main_ branch of amplify-codegen
+
+3. Move into your project folder
+
+```sh
 $ cd amplify-codegen
+```
+
+4. Run `setup-dev` script to installs dependencies and perform initial configuration. This command will also link a `amplify-dev` binary for your local testing.
 
-$ yarn run setup-dev
+```sh
+$ yarn setup-dev
 ```
 
-# Bugs
+> NOTE: The `amplify-dev` binary is built based on the latest amplify cli from npm registry and your local codegen packages. All your local changes from codegen can be reflected (typescript files need to be build by `tsc`). In addition, if you are a developer of cli repo, you can run the same command to override the `amplify-dev` binary.
+
+### Architecture of the codebase
+
+Amplify Codegen is a monorepo built with [Yarn Workspaces](https://yarnpkg.com/features/workspaces) and [Lerna](https://github.com/lerna/lerna). All packages live within the `packages/` directory in the root. Each category inside packages has its own `src/` and `package.json`.
+
+### Steps towards contribution
+
+- Make changes to required file.
+- Write unit tests
+- Yarn build
+- Run test suite
+- Test in sample app using [amplify-dev](#tests)
```

Comment:
should the link be `#Tests` ? Also include that link in above point `Run test suite`